agent/taskresource/grpcclient/credentialsfetcherclient.go

```diff
@@ -0,0 +1,122 @@
+package grpcclient
+
+import (
+	"context"
+	"os"
+	"time"
+
+	pb "github.com/aws/amazon-ecs-agent/agent/taskresource/grpcclient/credentialsfetcher"
+	"github.com/cihub/seelog"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type CredentialsFetcherClient struct {
+	conn    *grpc.ClientConn
+	timeout time.Duration
+}
+
+// GetGrpcClientConnection() returns grpc client connection
+func GetGrpcClientConnection() (*grpc.ClientConn, error) {
```

Comment:
Have 2 questions for this file,
Q. Are function `GetGrpcClientConnection()`, `GetSocketAddress()`, `NewCredentialsFetcherClient()`, `AddKerberosLease()`, and `DeleteKerberosLease()` all needed to be public methods? or can be private methods as them are called/used within the same `grpcclient` package.
  
Q. Where will `GetGrpcClientConnection()` be called/used? I have not found any usage in this PR and the [previous PR](https://github.com/aws/amazon-ecs-agent/pull/3406/files).

Reply:
This is just a template code for credentials-fetcher to pull gRPC to vendor, In the subsequent PR will test and update the access modifiers accordingly.